Usage of Deprecated getDependencyProject() Method in dev.iurysouza.modulegraph

The ProjectDependency.getDependencyProject() method is deprecated and is currently being used in the dev.iurysouza.modulegraph library. This results in multiple deprecation warnings during the Gradle build process.


‍`The ProjectDependency.getDependencyProject() method has been deprecated.`

💡 Suggested Fix:
Update the library to use the recommended replacement API for getDependencyProject(). Refer to the Gradle API documentation for the latest guidance.

✅ Why It Matters:
Continuing to use deprecated APIs can lead to build instability and future incompatibilities with Gradle versions. Updating ensures long-term compatibility and cleaner build output.

@iurysza 

Kotlin 2.2.0
AGP 8.12.0
moduleGraph 0.12.0
gradleWrapper 9.0.0

When running `./gradlew createModuleGraph` command, getting below exception while generating module graph for a project.

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'astro-mobile'.
> Could not create task ':createModuleGraph'.
   > 'org.gradle.api.Project org.gradle.api.artifacts.ProjectDependency.getDependencyProject()'
```

Do you planning to publish a new version with recommended gradle api usage.

--- a/.gitignore
+++ b/.gitignore
@@ -51,3 +51,5 @@ captures/
 .idea/jarRepositories.xml
 .idea/navEditor.xml
 !/gradle/wrapper/gradle-wrapper.jar
+.serena/
+CLAUDE.md

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.11.1-all.zip
 networkTimeout=10000
 validateDistributionUrl=true
 zipStoreBase=GRADLE_USER_HOME

--- a/plugin-build/modulegraph/src/main/kotlin/dev/iurysouza/modulegraph/gradle/graphparser/projectquerier/GradleProjectQuerier.kt
+++ b/plugin-build/modulegraph/src/main/kotlin/dev/iurysouza/modulegraph/gradle/graphparser/projectquerier/GradleProjectQuerier.kt
@@ -21,7 +21,7 @@ internal class GradleProjectQuerier(private val allProjects: Set<Project>) : Pro
         return project.configurations.map { config ->
             val projectPaths = config.dependencies
                 .withType(ProjectDependency::class.java)
-                .map { subproject -> subproject.dependencyProject.path }
+                .map { subproject -> subproject.path }
             GradleProjectConfiguration(config.name, projectPaths)
         }
     }

